src/operator/race-static.ts

```diff
@@ -8,14 +8,14 @@ import {isArray} from '../util/isArray';
  * @param {...Observables} ...observables sources used to race for which Observable emits first.
  * @returns {Observable} an Observable that mirrors the output of the first Observable to emit an item.
  */
-export function race<T>(...observables: Array<Observable<any> | Array<Observable<any>>>): Observable<T> {
+export function race<T>(...observables: Array<Observable<T> | Array<Observable<T>>>): Observable<T> {
   // if the only argument is an array, it was most likely called with
   // `pair([obs1, obs2, ...])`
   if (observables.length === 1) {
-    if (isArray(observables[0])) {
-      observables = <Array<Observable<any>>>observables[0];
-    } else {
+    if (!isArray(observables[0])) {
```

Comment:
Personally I don't have strong preferences for this, but would like to remain previous here. I do understand your intention and it makes sense, also it brings reversed conditional check flow (check `if x===true` to `if !x`). 

My take in this change is, let's PR to have specific scope and take this as separate topics if necessary.